File: Cli_Core_JavaFX/src/cli_core_javafx_test/Cmd_Item_Rem.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package cli_core_javafx_test;

/**
 *
 * @author mike
 */
public class Cmd_Item_Rem extends Cmd_Item_Base {

    public Cmd_Item_Rem(String str_rem, String help) {
        super(str_rem, help);
        Type = "Rem";
    }

    public String Debug_Value_Get() {
        return Text + " - rem";
    }

    public Cmd_Item_Valid_Result Parse(String s) {
        Value_Str = s;

        if (s.length() == 0) {
            return Cmd_Item_Valid_Result.CMD_ITEM_EMPTY;
        }

        if (s.substring(0, Text.length()).equals(Text)) {
            return Cmd_Item_Valid_Result.CMD_ITEM_OK;
        }

        return Cmd_Item_Valid_Result.CMD_ITEM_ERROR;
    }

}

File: Cli_Core_CS/Cli_Core_CS/Cmd_Item_Rem.cs
using System;
using System.Collections.Generic;
using System.Text;

namespace Cli_Core_CS
{
    class Cmd_Item_Rem : Cmd_Item_Base
    {
        public Cmd_Item_Rem(string str_rem, string help) : base(str_rem, help)
        {
            Type = "Rem";
        }

        override public string Debug_Value_Get()
        {
            return Text + " - rem";
        }

        override public Cmd_Item_Valid_Result Parse(string s)
        {
            Value_Str = s;

            if (s.Length == 0) return Cmd_Item_Valid_Result.CMD_ITEM_EMPTY;

            if (s.Substring(0, Text.Length) == Text) return Cmd_Item_Valid_Result.CMD_ITEM_OK;

            return Cmd_Item_Valid_Result.CMD_ITEM_ERROR;
        }
    }
}
